packages/nimble-components/src/table/index.ts

```diff
@@ -66,15 +97,39 @@ export class Table<
 
         // Ignore any updates that occur prior to the TanStack table being initialized.
         if (this.tableInitialized) {
-            this.updateTableOptions({ data: this.data });
+            this.trySetData(this.data);
         }
     }
 
+    public checkValidity(): boolean {
+        return this.tableValidator.isValid();
+    }
+
+    private trySetData(newData: TData[]): void {
+        const areIdsValid = this.tableValidator.validateDataIds(
+            newData,
+            this.idFieldName
+        );
+        if (areIdsValid) {
+            this.updateTableOptions({ data: newData });
+        } else {
+            this.updateTableOptions({ data: [] });
+        }
+    }
+
+    private refreshRows(): void {
+        const rows = this.table.getRowModel().rows;
+        this.tableData = rows.map(row => {
+            return { data: row.original, id: row.id } as TableRowState<TData>;
+        });
```

Comment:
I'm assuming this type is actually compatible with TableRowState, instead of the `as` cast (which disables checks and just asserts) can we leverage the type?

```suggestion
        this.tableData = rows.map(row => {
            const rowState: TableRowState<TData> = { data: row.original, id: row.id };
            return rowState;
        });
```

Reply:
Your suggestion actually caught a legitimate issue with this line, so I'll fix that too.